lib/active_model/serializer/adapter/json/fragment_cache.rb

```diff
@@ -1,16 +1,6 @@
-require 'active_model/serializer/adapter/fragment_cache'
-module ActiveModel
-  class Serializer
-    class Adapter
-      class Json < Adapter
-        class FragmentCache
+class ActiveModel::Serializer::Adapter::Json::FragmentCache
```

Comment:
*(for reference, the commit message is here https://github.com/bf4/active_model_serializers/commit/cf6a074a1c62e3f5a30f90a0987b8cf049272953)*

To avoid the collision issue I think it's good to only use `module` for namespacing. I think that should avoid the issue?

I also think that automatic registration when subclassing is very magical. My guess is the common 'Rails-way' is to do on-the-fly lookups when an adapter is needed (constantize a string and look in some pre-defined directory, e.g. `app/serializers/adapters`) or to register them explicitly in an initializer.

But I'm no Rails-export, I may be wrong on the conventions. Perhaps @spastorino can weight in?